src/printer/components/projects.js

```diff
@@ -28,14 +31,38 @@ const metadata = {
  * @param {object} b
  */
 const sortByType = (a, b) => {
-  if (a.type == "folder" && b.type == "folder") {
-    return a.display.localeCompare(b.display);
+  if (a.type == b.type) {
+    let aName = a.display || a.name || "";
+    let bName = b.display || b.name || "";
+
+    return aName.localeCompare(bName);
   } else if (a.type == "folder") {
     return -1;
+  } else if (b.type == "folder") {
+    return 1;
   }
   return b.date - a.date;
 };
 
+const getInitialMetadataFiles = (data) => {
```

Comment:
indeed the files should not be separated, it's to be shown as they come.
- folder
- file1.gcode  (local)
- file1.gcode (sdcard)

Reply:
@filipkotoucek Is it that way? 
The files should not be separated from PCL

Reply:
@ondratu

Reply:
For SL1 yes. We combine both storages. It is historical reason and we want to keep it consistent with touch-ui. On the other hand it can be separated on API layer but UI itself can merge all files from all storages into one.

Reply:
And yes. The sorting order should be folders first (by date) and after that files (by date). So the newest folder is on top, and the newest file is right behind all folders.

Reply:
The octoprint doesn't show date in folders, could be the folders by name as the current touch UI?